compiler/passes/initializerRules.cpp

```diff
@@ -551,29 +551,29 @@ static InitNormalize preNormalize(AggregateType* at,
 
       // Stmt is assignment to a super field
       } else if (DefExpr* field = toSuperFieldInit(state.type(), callExpr)) {
-        if (state.isPhase2() == false) {
+        if (state.isPhase0() == true) {
           USR_FATAL(stmt,
                     "can't set value of field \"%s\" from parent type "
-                    "during phase 1 of initialization",
+                    "before calling this.init() or super.init()",
                     field->sym->name);
 
         } else {
           if (field->sym->hasFlag(FLAG_CONST) == true) {
             USR_FATAL(stmt,
                       "cannot update a const field, \"%s\", "
-                      "from parent type in phase 2",
+                      "from parent type outside of this.init() or super.init()",
```

Comment:
I don't find this error message particularly clear.  `this.init()` and `super.init()` are calls that are made.  Sending an argument to the `super.init()` or `this.init()` gives no guarantee that the value will be used for that particular field.

I think this should say "cannot update a const field, x, from parent type in child initializer" or something like that.

Reply:
Sounds good, I like your suggested wording better.